### DISCUSSION TITLE
TensorXf object copy and in-placing operation

### DISCUSSION THREAD

**Discussion by Jackson334-prog** [mitsuba-renderer/mitsuba3#977] [Q&A] [answered]
## Summary
Hi, I am working on how to apply some value for each channel of tensor , so I need to use the function per-RGB channel , seems like there is no in-placing operator in TensorXf object , so I can not simply assign just like `img[:,:,0] = var` such kind of thing , also I check for the `dr.copy` `dr.clone` none of these function. So I wonder how can I do some operating on different RGB channel and copy per channel into original TensorXf object.

<img width="538" alt="屏幕截图 2023-11-14 215931" src="https://github.com/mitsuba-renderer/mitsuba3/assets/73884949/f31432f1-e4dc-48f3-b779-57409f241db9">

**Comment by njroussel** [comment, marked as answer]
Hi @ljijjiang1234 

Assignment to tensor objects is done by calling `dr.scatter(tensor.array, new_value, indices)`. The `tensor.array` member is the underlying flat buffer that is used to store the buffer. It's essentially the result of `dr.ravel()`.

This interface should improve soon-_ish_.

**Reply by Jackson334-prog** [reply]
Holy thanks for your patient reply and your rigorous work , I have been solved for this problem by using `dr.scatter()`.
